homeassistant/components/statistics/sensor.py

```diff
@@ -166,6 +202,9 @@ def __init__(
         self.is_binary = self._source_entity_id.split(".")[0] == "binary_sensor"
         self._name = name
         self._state_characteristic = state_characteristic
+        if self._state_characteristic == STAT_DEFAULT:
+            self._state_characteristic = STAT_COUNT if self.is_binary else STAT_MEAN
+            _LOGGER.warning(DEPRECATION_WARNING, self._state_characteristic, name)
```

Comment:
We typically do this during schema validation instead, add it as the first validator in `PLATFORM_SCHEMA`.

Reply:
This way I might even be able to avoid the placeholder `STAT_DEFAULT`. Will look into it in the next few days.